src/support/timestamp.c

```diff
@@ -495,79 +510,57 @@ __wt_time_value_validate(
         WT_TIME_VALIDATE_RET(session,
           "value time window has a durable start time after its durable stop time; time window %s",
           __wt_time_window_to_string(tw, time_string[0]));
-
-    if (tw->prepare && F_ISSET(S2C(session), WT_CONN_PRESERVE_PREPARED)) {
-        if (!WT_TIME_WINDOW_HAS_START_PREPARE(tw) && !WT_TIME_WINDOW_HAS_STOP_PREPARE(tw)) {
+    /* Validate that if start prepare_ts is set, start_prepared_id must be set */
+    if (WT_TIME_WINDOW_HAS_START_PREPARE(tw)) {
+        if (tw->start_prepare_ts == WT_TS_NONE) {
+            WT_TIME_VALIDATE_RET(session,
+              "Start prepared value time window has no start prepare time "
+              "window %s",
+              __wt_time_window_to_string(tw, time_string[0]));
+        }
+        if (tw->start_prepared_id == WT_PREPARED_ID_NONE &&
+          F_ISSET(S2C(session), WT_CONN_PRESERVE_PREPARED)) {
```

Comment:
In the non preserve prepared case, can we validate it as none?

Reply:
done